include/wx/aui/auibar.h

```diff
@@ -244,7 +249,8 @@ class WXDLLIMPEXP_AUI wxAuiToolBarItem
     bool m_active;               // true if the item is currently active
     bool m_dropDown;             // true if the item has a dropdown button
     bool m_sticky;               // overrides button states if true (always active)
-    long m_userData;            // user-specified data
+    long m_userData;             // user-specified data
+    wxObject* m_clientData;      // client data
```

Comment:
It looks like they're supposed to be used for the same thing and I'd at least describe them better, e.g.
```suggestion
    long m_userData;             // numeric user-specified data
    wxObject* m_clientData;      // pointer to user-specified data
```

But it's still somewhat of a mess :-(

Reply:
What about: "alternative client data, compatible with wxToolBar" for the wxObject* 

Just some feedback on the status quo from a user perspective: I am using those long values as workaround as the orginal `m_clientData` does not work and it works fine. 
There is indeed potential confusion if both exist and they are not the same, but imho the alternative is even more confusing. As long as this is mentioned in the documentation this is one of the more easy traps to master when using AUI, like you can not manage wxFrame or wxDialog derived stuff its just crashing (wasn't obvious to me). There are other things confusing and annoying like different named functions doing the same thing, e.g. `aui::GetToolCount() `and `wx::GetToolsCount()`. But some things are better with AUI, e.g. there is actually a documented Tool class. wxToolBarToolBase is not documented at all (using 3.0.5). 
I just wrote a common base class to those two Toolbars for own additions and I had introduce about 10 virtual functions that were basically doing the same thing but slightly different between the two.

Reply:
FWIW we could also change the type of `m_userData` to `wxUIntPtr` and use it to store either long or a pointer. This would be type-unsafe, of course, but then so is using `wxObject*` in the first place anyhow...